examples/kubernetes/connectivity-check/services.cue

```diff
@@ -0,0 +1,38 @@
+package connectivity_check
+
+deployment: [ID=_]: {
```

Comment:
Any particular reason why this is not placed in `defaults.cue`?

In general, I think that minimizing the places where an entity (in this case `deployment`) is defined might be a good way to strike a good balance between making it easy to write and easy to understand/modify.

Reply:
I tried both approaches, and I didn't conclusively decide one path as superior to the other. Either way, the core definitions are in yet another file (`resources.cue`) and yet in this particular case the definitions are only relevant to this file.

But at the same time, these *will* apply globally so perhaps it's better to shift them into defaults.go. Another option is also to combine defaults.go and resources.go to put all of the code in one place (though I *think* it might be slightly clearer to have defaults.go be smaller and more concise for the type of defaults they provide).

Reply:
I don't feel strongly about either case, so up to you.

How to factor definitions seems like a non-trivial aspect of cue. While the connectivity tests are fairly self-contained and I guess it does not matter much, it's a good playground to experiment and see if and how to extend cue's use.

Reply:
I've added a new commit near the end to document the layout and how to interpret the files.